### PR TITLE
Trapping Rainwater in C++

### PR DESCRIPTION
**Pull request by Rishika30**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

I have used two pointer approach to solve the trapping rainwater problem in cpp. This is the optimised solution which takes O(n) time and O(1) space.  In two pointer approach, we take the starting index and the last index and start iterating comparing them.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2873

### CODE REVIEW

File: C-Plus-Plus/other/trappingRainwater.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/* C++ program to solve Trapping Rainwater problem.An array of non-negative integers are given which
+represent an elevation height each of width 1, the
+goal is to compute total water trapped after raining*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+class Solution {
```
Comment (HarshCasper) [inline C-Plus-Plus/other/trappingRainwater.cpp]: Use meaningful class names

File: C-Plus-Plus/other/trappingRainwater.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Explanation -
+
+Two units of water can be stored at indexe 2 and 
+3 units at index 3
+
+Time Complexity : O(n)
+Space Complexity : O(1) 
+
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/other/trappingRainwater.cpp]: Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format and Indent your Code properly. Add Markdown changes as well.

File: C-Plus-Plus/other/trappingRainwater.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        		else
+        			res += max_right - height[right];
+        		right--;
+        	}
+        	return res;
+    }
+};
+
+int main(){
+	int n, val;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/trappingRainwater.cpp]: Use suitable naming convention

File: C-Plus-Plus/other/trappingRainwater.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	vector<int> height;
+	cin >> n;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/trappingRainwater.cpp:47]: Display a message what are you taking input from the user before this line

File: C-Plus-Plus/other/trappingRainwater.cpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	while(n--){
+		cin >> val;
+		height.push_back(val);
+	}
+	Solution sol;
+	int result = sol.trap(height);
+	cout << result;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/trappingRainwater.cpp:56]: Display message what are you giving output for better user-interface

File: C-Plus-Plus/other/trappingRainwater.cpp [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+/*
+
+Input -
+
+n = 5
+arr[n] = {2,3,1,0,5}
+
+Sample Output -
+
+Maximum water that can be accumulated is 5
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/trappingRainwater.cpp]: You are not displaying any message, then how you are writing the same in Sample Output.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Rishika30 Do the suggested changes & Update the **Readme** file details

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make README changes as well

**Comment by siddharth25pandey** [comment]
@Rishika30 Resolve the merge conflicts

**Pull request merged**